src/taskgraph/decision.py

```diff
@@ -264,6 +272,38 @@ def _determine_more_accurate_base_ref(repo, candidate_base_ref, head_ref, base_r
     return base_ref
 
 
+def _determine_more_accurate_base_rev(repo, base_ref, candidate_base_rev, head_rev):
+    base_ref_or_rev = candidate_base_rev
+    if not candidate_base_rev:
+        logger.warning("base_rev is not set.")
+        base_ref_or_rev = base_ref
+    elif candidate_base_rev == Repository.NULL_REVISION:
+        logger.warning("base_rev equals the null revision. This branch is a new one.")
```

Comment:
I'm a bit worried that logging all these at the warning level is going to lead people to believe something went wrong. These feel more like `debug` messages to me (i.e, we'd only really want to see them when something *has* gone wrong and we've enabled extra logging to determine what. Though I could probably be convinced to leave them at the info level.

Reply:
You're right, maybe I was too easygoing with warning logs here. That said, I believe they have some value being output as INFO.

Let's say a developer doesn't get the expected tasks in a graph: either they or we have a way to immediately look into the logs and see what happened in this case. That doesn't require anyone to try to reproduce locally. Especially given the fact that the generated `parameters.yml` won't let you reproduce the bug since it contains the new `base_rev`/`base_ref` values. 

In addition to that, neither the developer nor us will likely remember to `git fetch` the `base_rev` Github provides. This means, the behavior will be different by default, locally. 

Finally, pushing things (to turn on debug logs, for instance) may lead to a different state which makes the bug not reproducible remotely either. 

Long story short: there are many footguns which will take time to figure out. Let's save ourselves some time by always being explicit about these heuristics.

What do you think?

Reply:
My only counter arguments are:
1. We log all the detailed optimization stuff at the DEBUG level already, so people kind of need to use DEBUG anyway to figure out why their task isn't running (though these logs are massive so there's good reason to hide them by default).
2. Gecko decision tasks use DEBUG verbosity by default, other repos could too.

But yeah, logging at INFO sounds fine to me.. no compelling reason not to.

Reply:
I agree, we should switch to DEBUG level anyway to get into the optimization stuff. That said, that part is stateless unlike the `base_rev`/`base_ref` resets. I'd better be safe than sorry in this case. I really how confused people (myself included) can be because of the state of the repo.